codalab/lib/bundle_cli.py

```diff
@@ -1028,81 +1049,33 @@ def copy_bundle(self, source_client, source_bundle_uuid, dest_client, dest_works
-                filename_suffix = '.gz'
-                if isinstance(source_client, LocalBundleClient):
-                    source = source_client.open_gzipped_file(target)
-                else:
-                    source = RPCFileHandle(
-                        source_client.open_gzipped_file(target),
-                        source_client.proxy, finalize_on_close=True)
-
-            if source is not None:
-                # Open target (temporary file)
-                if isinstance(dest_client, LocalBundleClient):
-                    dest_path = tempfile.mkstemp(filename_suffix)[1]
-                    dest = open(dest_path, 'wb')
-                else:
-                    dest_file_uuid = dest_client.open_temp_file('bundle' + filename_suffix)
-                    dest = RPCFileHandle(dest_file_uuid, dest_client.proxy)
-                with closing(dest):
-                    # Copy contents over from source to target.
-                    file_util.copy(
-                        source,
-                        dest,
-                        autoflush=False,
-                        print_status='Copying %s from %s to %s' % (source_bundle_uuid, source_client.address, dest_client.address))
-
-            # Set sources
-            if source is None:
-                sources = None
-            elif isinstance(dest_client, LocalBundleClient):
-                sources = [dest_path]
-            else:
-                sources = [dest_file_uuid]
-
-            # Finally, install the archive (this function will delete it).
-            if isinstance(dest_client, LocalBundleClient):
-                result = dest_client.upload_bundle(
-                    sources=sources,
-                    follow_symlinks=False,
-                    exclude_patterns=None,
-                    git=False,
-                    unpack=True,
-                    remove_sources=True,
-                    info=source_info,
-                    worksheet_uuid=dest_worksheet_uuid,
-                    add_to_worksheet=add_to_worksheet,
-                )
-            else:
-                result = dest_client.finish_upload_bundle(
-                    sources,
-                    True,
-                    source_info,
-                    dest_worksheet_uuid,
-                    add_to_worksheet)
-            
-            return result
-        except:
-            if source is not None:
-                source.close()
-            if dest_file_uuid is not None:
-                dest_client.finalize_file(dest_file_uuid)
-            raise
+        target_info = source_client.fetch_contents_info(*target)
+
+        # Create the bundle, copying over metadata from the source bundle
+        dest_bundle = dest_client.create('bundles', source_info, params={
+            'worksheet': dest_worksheet_uuid,
+            'detached': not add_to_worksheet,
+        })
+
+        if target_info is None:
```

Comment:
So it turns out that this actually does happen, in the case where the bundle's contents have been deleted (using `cl rm -d`), `test-cli.py` has a check specifically for this case.